### PR TITLE
Fix missing masterPublicURL on console configmap

### PR DESCRIPTION
**Pull request by benjaminapetersen**
bug 1664574
([link](https://bugzilla.redhat.com/show_bug.cgi?id=1664574))
- Add `infrastructureConfig` to supply console `configmap` with `masterPublicURL`.

This will likely conflict with #162

### CODE REVIEW

**Comment by benjaminapetersen** [comment]
Testing yet, my cluster fell over.

**Comment by spadgett** [comment]
CI failures look legitimate

**Comment by benjaminapetersen** [comment]
rbac updated

File: pkg/console/operator/operator.go [hunk 1]
```diff
@@ -27,7 +27,8 @@ import (
 	"github.com/openshift/library-go/pkg/operator/events"
 
 	// informers
-	configinformerv1 "github.com/openshift/client-go/config/informers/externalversions/config/v1"
+	configinformer "github.com/openshift/client-go/config/informers/externalversions"
+	// configinformerv1 "github.com/openshift/client-go/config/informers/externalversions/config/v1"
```
Comment (spadgett) [inline pkg/console/operator/operator.go]: Remove commented out code

File: manifests/03-rbac-role.yaml [hunk 2]
```diff
@@ -122,6 +122,14 @@ rules:
   - watch
   resourceNames:
   - console
+- apiGroups:
+  - config.openshift.io
+  resources:
+  - infrastructures
```
Comment (spadgett) [inline manifests/03-rbac-role.yaml]: I'd combine this with the existing rule for api group `config.openshift.io`

File: pkg/console/operator/operator.go [hunk 3]
```diff
@@ -148,15 +162,22 @@ func (c *consoleOperator) Sync(obj metav1.Object) error {
 		}
 	}
 
-	if err := c.handleSync(operatorConfig, consoleConfig); err != nil {
+	// we need infrastructure config for apiServerURL.
+	// it is helpful but not essential so we should not die if its not available.
```
Comment (spadgett) [inline pkg/console/operator/operator.go]: Is there ever a time we'd expect this not to work? I feel like we should just fail here. Public URL is needed for a few things in console.

Comment (benjaminapetersen) [reply pkg/console/operator/operator.go]: Well, I figure we need it only for a few small things, that we can guard in console code. So console can be 99% useful without it. Once it's available, the new pods will roll out anyway.

Comment (spadgett) [reply pkg/console/operator/operator.go]: It should always be available right? The drawback is that we're not reporting an error anywhere (except in the operator log). You won't know something is broken.

File: manifests/03-rbac-role.yaml (hunk 2 above)
Comment (benjaminapetersen) [reply manifests/03-rbac-role.yaml]: We don't want to `update`, `delete`, `create`. 
Just watch rules.

File: pkg/console/operator/operator.go (hunk 3 above)
Comment (benjaminapetersen) [reply pkg/console/operator/operator.go]: Ill update.

File: pkg/console/operator/operator.go (hunk 1 above)
Comment (benjaminapetersen) [reply pkg/console/operator/operator.go]: will do.

File: manifests/03-rbac-role.yaml (hunk 2 above)
Comment (spadgett) [reply manifests/03-rbac-role.yaml]: Thanks, missed that. Looks good.

File: pkg/console/subresource/configmap/configmap_test.go [hunk 4]
```diff
@@ -61,6 +65,11 @@ func TestDefaultConfigMap(t *testing.T) {
 					Spec:   configv1.ConsoleSpec{},
 					Status: configv1.ConsoleStatus{},
 				},
+				infrastructureConfig: &configv1.Infrastructure{
+					Status: configv1.InfrastructureStatus{
+						APIServerURL: "https://api.some.cluster.openshift.com:6443",
```
Comment (spadgett) [inline pkg/console/subresource/configmap/configmap_test.go]: should probably use the const

File: pkg/console/starter/starter.go [hunk 5]
```diff
@@ -127,23 +127,23 @@ func RunOperator(ctx *controllercmd.ControllerContext) error {
...
 	}
 
 	// TODO: rearrange these into informer,client pairs, NOT separated.
 	consoleOperator := operator.NewConsoleOperator(
 		// informers
-		consoleOperatorConfigInformers.Operator().V1().Consoles(), // OperatorConfig
-		consoleConfigInformers.Config().V1().Consoles(),           // ConsoleConfig
-
+		operatorConfigInformers.Operator().V1().Consoles(), // OperatorConfig
+		// configInformers.Config().V1().Consoles(),           // ConsoleConfig
```
Comment (spadgett) [inline pkg/console/starter/starter.go]: remove commented out code?

File: pkg/console/subresource/configmap/configmap_test.go (hunk 4 above)
Comment (benjaminapetersen) [reply pkg/console/subresource/configmap/configmap_test.go]: missed that, will follow-up.

**Pull request merged**